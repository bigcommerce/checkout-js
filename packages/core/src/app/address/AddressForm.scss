@import "../ui/Base";

.checkout-address:has(.floating-form-field) {
    margin-left: -0.5 * $floating-form-field-spacing;
    margin-right: -0.5 * $floating-form-field-spacing;
}

.checkout-address {
    display: flex;
    flex-flow: row wrap;
    margin-left: -1 * spacing("half");
    margin-right: -1 * spacing("half");

    .dynamic-form-addressState,
    .dynamic-form-field {
        display: block;
        flex: 1;
        flex-basis: 100%;
        margin-bottom: spacing("half");
        max-width: 100%;
        padding: 0 spacing("half");

        &:last-child {
            margin-bottom: 0;
        }
    }

    .dynamic-form-field.floating-form-field {
        margin-bottom: $floating-form-field-spacing;
        padding: 0 0.5 * $floating-form-field-spacing;
    }

    .dynamic-form-field--province,
    .dynamic-form-field--provinceCode,
    .dynamic-form-field--firstName,
    .dynamic-form-field--lastName,
    .dynamic-form-field--phone,
    .dynamic-form-field--city {
        @include breakpoint("small") {
            flex-basis: 50%;
        }
    }

    .dynamic-form-field--postCode {
        @include breakpoint("small") {
            flex-basis: 30%;
        }
    }
}
