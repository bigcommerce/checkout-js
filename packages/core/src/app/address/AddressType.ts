enum AddressType {
    Billing = 'billing',
    Shipping = 'shipping',
}

export default AddressType;
