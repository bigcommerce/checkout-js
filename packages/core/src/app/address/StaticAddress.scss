.checkout-address--static {
    .address-entry {
        margin: 0;
    }
}
