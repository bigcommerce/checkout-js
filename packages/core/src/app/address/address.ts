export interface AddressKeyMap<T = string> {
    [fieldName: string]: T;
}
