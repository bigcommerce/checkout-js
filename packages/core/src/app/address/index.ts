export { default as mapAddressToFormValues, AddressFormValues } from './mapAddressToFormValues';
export { default as mapAddressFromFormValues } from './mapAddressFromFormValues';
export { default as AddressForm } from './AddressForm';
export { default as AddressFormModal } from './AddressFormModal';
export { default as AddressSelect } from './AddressSelect';
export { default as AddressType } from './AddressType';
export { default as StaticAddress } from './StaticAddress';
export { default as localizeAddress } from './localizeAddress';
export { default as isValidAddress } from './isValidAddress';
export { default as isValidCustomerAddress } from './isValidCustomerAddress';
export { default as isEqualAddress } from './isEqualAddress';
export {
    default as getAddressFormFieldsValidationSchema,
    getTranslateAddressError,
} from './getAddressFormFieldsValidationSchema';
