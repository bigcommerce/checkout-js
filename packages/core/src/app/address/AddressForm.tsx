import { Address, Country, FormField } from '@bigcommerce/checkout-sdk';
import { memoize } from '@bigcommerce/memoize';
import { forIn, noop } from 'lodash';
import React, { Component, createRef, ReactNode, RefObject } from 'react';

import { TranslatedString, withLanguage, WithLanguageProps } from '../locale';
import { AutocompleteItem } from '../ui/autocomplete';
import { CheckboxFormField, DynamicFormField, DynamicFormFieldType, Fieldset } from '../ui/form';

import { AddressKeyMap } from './address';
import {
    getAddressFormFieldInputId,
    getAddressFormFieldLegacyName,
} from './getAddressFormFieldInputId';
import { GoogleAutocompleteFormField, mapToAddress } from './googleAutocomplete';
import './AddressForm.scss';

export interface AddressFormProps {
    fieldName?: string;
    countryCode?: string;
    countriesWithAutocomplete?: string[];
    countries?: Country[];
    formFields: FormField[];
    googleMapsApiKey?: string;
    shouldShowSaveAddress?: boolean;
    useFloatingLabel?: boolean;
    onAutocompleteSelect?(address: Partial<Address>): void;
    onAutocompleteToggle?(state: { inputValue: string; isOpen: boolean }): void;
    onChange?(fieldName: string, value: string | string[]): void;
    setFieldValue?(fieldName: string, value: string | string[]): void;
}

const LABEL: AddressKeyMap = {
    address1: 'address.address_line_1_label',
    address2: 'address.address_line_2_label',
    city: 'address.city_label',
    company: 'address.company_name_label',
    countryCode: 'address.country_label',
    firstName: 'address.first_name_label',
    lastName: 'address.last_name_label',
    phone: 'address.phone_number_label',
    postalCode: 'address.postal_code_label',
    stateOrProvince: 'address.state_label',
    stateOrProvinceCode: 'address.state_label',
};

const AUTOCOMPLETE: AddressKeyMap = {
    address1: 'address-line1',
    address2: 'address-line2',
    city: 'address-level2',
    company: 'organization',
    countryCode: 'country',
    firstName: 'given-name',
    lastName: 'family-name',
    phone: 'tel',
    postalCode: 'postal-code',
    stateOrProvince: 'address-level1',
    stateOrProvinceCode: 'address-level1',
};

const PLACEHOLDER: AddressKeyMap = {
    countryCode: 'address.select_country_action',
    stateOrProvince: 'address.select_state_action',
    stateOrProvinceCode: 'address.select_state_action',
};

const AUTOCOMPLETE_FIELD_NAME = 'address1';

class AddressForm extends Component<AddressFormProps & WithLanguageProps> {
    private containerRef: RefObject<HTMLElement> = createRef();
    private nextElement?: HTMLElement | null;

    private handleDynamicFormFieldChange: (name: string) => (value: string | string[]) => void =
        memoize((name) => (value) => {
            this.syncNonFormikValue(name, value);
        });

    componentDidMount(): void {
        const { current } = this.containerRef;

        if (current) {
            this.nextElement = current.querySelector<HTMLElement>('[autocomplete="address-line2"]');
        }
    }

    render(): ReactNode {
        const {
            formFields,
            fieldName,
            countriesWithAutocomplete,
            countryCode,
            googleMapsApiKey,
            onAutocompleteToggle,
            shouldShowSaveAddress,
            useFloatingLabel,
        } = this.props;

        return (
            <>
                <Fieldset>
                    <div
                        className="checkout-address"
                        ref={this.containerRef as RefObject<HTMLDivElement>}
                    >
                        {formFields.map((field) => {
                            const addressFieldName = field.name;
                            const translatedPlaceholderId = PLACEHOLDER[addressFieldName];

                            if (
                                addressFieldName === 'address1' &&
                                googleMapsApiKey &&
                                countriesWithAutocomplete
                            ) {
                                return (
                                    <GoogleAutocompleteFormField
                                        apiKey={googleMapsApiKey}
                                        countryCode={countryCode}
                                        field={field}
                                        key={field.id}
                                        nextElement={this.nextElement || undefined}
                                        onChange={this.handleAutocompleteChange}
                                        onSelect={this.handleAutocompleteSelect}
                                        onToggleOpen={onAutocompleteToggle}
                                        parentFieldName={fieldName}
                                        supportedCountries={countriesWithAutocomplete}
                                        useFloatingLabel={useFloatingLabel}
                                    />
                                );
                            }

                            return (
                                <DynamicFormField
                                    autocomplete={AUTOCOMPLETE[field.name]}
                                    extraClass={`dynamic-form-field--${getAddressFormFieldLegacyName(
                                        addressFieldName,
                                    )}`}
                                    field={field}
                                    inputId={getAddressFormFieldInputId(addressFieldName)}
                                    // stateOrProvince can sometimes be a dropdown or input, so relying on id is not sufficient
                                    key={`${field.id}-${field.name}`}
                                    label={
                                        field.custom ? (
                                            field.label
                                        ) : (
                                            <TranslatedString id={LABEL[field.name]} />
                                        )
                                    }
                                    onChange={this.handleDynamicFormFieldChange(addressFieldName)}
                                    parentFieldName={
                                        field.custom
                                            ? fieldName
                                                ? `${fieldName}.customFields`
                                                : 'customFields'
                                            : fieldName
                                    }
                                    placeholder={this.getPlaceholderValue(
                                        field,
                                        translatedPlaceholderId,
                                    )}
                                    useFloatingLabel={useFloatingLabel}
                                />
                            );
                        })}
                    </div>
                </Fieldset>
                {shouldShowSaveAddress && (
                    <CheckboxFormField
                        labelContent={<TranslatedString id="address.save_in_addressbook" />}
                        name={fieldName ? `${fieldName}.shouldSaveAddress` : 'shouldSaveAddress'}
                    />
                )}
            </>
        );
    }

    private getPlaceholderValue(field: FormField, translatedPlaceholderId: string): string {
        const { language } = this.props;

        if (field.default && field.fieldType !== 'dropdown') {
            return field.default;
        }

        return translatedPlaceholderId && language.translate(translatedPlaceholderId);
    }

    private handleAutocompleteChange: (value: string, isOpen: boolean) => void = (
        value,
        isOpen,
    ) => {
        if (!isOpen) {
            this.syncNonFormikValue(AUTOCOMPLETE_FIELD_NAME, value);
        }
    };

    private handleAutocompleteSelect: (
        place: google.maps.places.PlaceResult,
        item: AutocompleteItem,
    ) => void = (place, { value: autocompleteValue }) => {
        const { countries, setFieldValue = noop, onChange = noop } = this.props;

        const address = mapToAddress(place, countries);

        forIn(address, (value, fieldName) => {
            setFieldValue(fieldName, value as string);
            onChange(fieldName, value as string);
        });

        if (autocompleteValue) {
            this.syncNonFormikValue(AUTOCOMPLETE_FIELD_NAME, autocompleteValue);
        }
    };

    // because autocomplete state is controlled by Downshift, we need to manually keep formik
    // value in sync when autocomplete value changes
    private syncNonFormikValue: (fieldName: string, value: string | string[]) => void = (
        fieldName,
        value,
    ) => {
        const { formFields, setFieldValue = noop, onChange = noop } = this.props;

        const dateFormFieldNames = formFields
            .filter((field) => field.custom && field.fieldType === DynamicFormFieldType.date)
            .map((field) => field.name);

        if (fieldName === AUTOCOMPLETE_FIELD_NAME || dateFormFieldNames.indexOf(fieldName) > -1) {
            setFieldValue(fieldName, value);
        }

        onChange(fieldName, value);
    };
}

export default withLanguage(AddressForm);
