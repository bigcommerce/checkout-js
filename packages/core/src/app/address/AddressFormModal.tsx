import { Country, FormField } from '@bigcommerce/checkout-sdk';
import { FormikProps, withFormik } from 'formik';
import React, { FunctionComponent } from 'react';
import { lazy } from 'yup';

import { preventDefault } from '../common/dom';
import { TranslatedString, withLanguage, WithLanguageProps } from '../locale';
import { Button, ButtonVariant } from '../ui/button';
import { Form } from '../ui/form';
import { LoadingOverlay } from '../ui/loading';
import { Modal, ModalHeader } from '../ui/modal';

import AddressForm from './AddressForm';
import getAddressFormFieldsValidationSchema from './getAddressFormFieldsValidationSchema';
import { AddressFormValues } from './mapAddressToFormValues';

export interface AddressFormModalProps extends AddressFormProps {
    isOpen: boolean;
    onAfterOpen?(): void;
}

export interface AddressFormProps {
    countries?: Country[];
    countriesWithAutocomplete: string[];
    googleMapsApiKey?: string;
    isLoading: boolean;
    shouldShowSaveAddress?: boolean;
    defaultCountryCode?: string;
    useFloatingLabel?: boolean;
    getFields(countryCode?: string): FormField[];
    onSaveAddress(address: AddressFormValues): void;
    onRequestClose?(): void;
}

const SaveAddress: FunctionComponent<
    AddressFormProps & WithLanguageProps & FormikProps<AddressFormValues>
> = ({
    googleMapsApiKey,
    getFields,
    countriesWithAutocomplete,
    countries,
    values,
    setFieldValue,
    isLoading,
    onRequestClose,
    useFloatingLabel,
}) => (
    <Form autoComplete="on">
        <LoadingOverlay isLoading={isLoading}>
            <AddressForm
                countries={countries}
                countriesWithAutocomplete={countriesWithAutocomplete}
                countryCode={values.countryCode}
                formFields={getFields(values.countryCode)}
                googleMapsApiKey={googleMapsApiKey}
                setFieldValue={setFieldValue}
                shouldShowSaveAddress={false}
                useFloatingLabel={useFloatingLabel}
            />
            <div className="form-actions">
                <a
                    className="button optimizedCheckout-buttonSecondary"
                    href="#"
                    onClick={preventDefault(onRequestClose)}
                >
                    <TranslatedString id="common.cancel_action" />
                </a>

                <Button
                    disabled={isLoading}
                    id="checkout-save-address"
                    type="submit"
                    variant={ButtonVariant.Primary}
                >
                    <TranslatedString id="address.save_address_action" />
                </Button>
            </div>
        </LoadingOverlay>
    </Form>
);

const SaveAddressForm = withLanguage(
    withFormik<AddressFormProps & WithLanguageProps, AddressFormValues>({
        handleSubmit: (values, { props: { onSaveAddress } }) => {
            onSaveAddress(values);
        },
        mapPropsToValues: ({ defaultCountryCode = '' }) => ({
            firstName: '',
            lastName: '',
            address1: '',
            address2: '',
            customFields: {},
            country: '',
            countryCode: defaultCountryCode,
            stateOrProvince: '',
            stateOrProvinceCode: '',
            postalCode: '',
            phone: '',
            city: '',
            company: '',
            shouldSaveAddress: false,
        }),
        validationSchema: ({ language, getFields }: AddressFormProps & WithLanguageProps) =>
            lazy<Partial<AddressFormValues>>((values) =>
                getAddressFormFieldsValidationSchema({
                    language,
                    formFields: getFields(values && values.countryCode),
                }),
            ),
    })(SaveAddress),
);

const AddressFormModal: FunctionComponent<AddressFormModalProps> = ({
    isOpen,
    onAfterOpen,
    onRequestClose,
    ...addressFormProps
}) => (
    <Modal
        additionalModalClassName="modal--medium"
        header={
            <ModalHeader>
                <TranslatedString id="address.add_address_heading" />
            </ModalHeader>
        }
        isOpen={isOpen}
        onAfterOpen={onAfterOpen}
        onRequestClose={onRequestClose}
        shouldShowCloseButton={true}
    >
        <SaveAddressForm {...addressFormProps} onRequestClose={onRequestClose} />
    </Modal>
);

export default AddressFormModal;
