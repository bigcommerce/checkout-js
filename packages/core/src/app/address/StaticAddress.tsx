import {
    Address,
    CheckoutSelectors,
    Country,
    FormField,
    ShippingInitializeOptions,
} from '@bigcommerce/checkout-sdk';
import { isEmpty } from 'lodash';
import React, { FunctionComponent, memo } from 'react';

import { CheckoutContextProps, withCheckout } from '../checkout';

import AddressType from './AddressType';
import isValidAddress from './isValidAddress';
import localizeAddress from './localizeAddress';
import './StaticAddress.scss';

export interface StaticAddressProps {
    address: Address;
    type?: AddressType;
}

export interface StaticAddressEditableProps extends StaticAddressProps {
    initialize?(options: ShippingInitializeOptions): Promise<CheckoutSelectors>;
}

interface WithCheckoutStaticAddressProps {
    countries?: Country[];
    fields?: FormField[];
}

const StaticAddress: FunctionComponent<
    StaticAddressEditableProps & WithCheckoutStaticAddressProps
> = ({ countries, fields, address: addressWithoutLocalization }) => {
    const address = localizeAddress(addressWithoutLocalization, countries);
    const isValid = !fields
        ? !isEmpty(address)
        : isValidAddress(
              address,
              fields.filter((field) => !field.custom),
          );

    return !isValid ? null : (
        <div className="vcard checkout-address--static">
            {(address.firstName || address.lastName) && (
                <p className="fn address-entry">
                    <span className="first-name">{`${address.firstName} `}</span>
                    <span className="family-name">{address.lastName}</span>
                </p>
            )}

            {(address.phone || address.company) && (
                <p className="address-entry">
                    <span className="company-name">{`${address.company} `}</span>
                    <span className="tel">{address.phone}</span>
                </p>
            )}

            <div className="adr">
                <p className="street-address address-entry">
                    <span className="address-line-1">{`${address.address1} `}</span>
                    {address.address2 && (
                        <span className="address-line-2">{` / ${address.address2}`}</span>
                    )}
                </p>

                <p className="address-entry">
                    {address.city && <span className="locality">{`${address.city}, `}</span>}
                    {address.localizedProvince && (
                        <span className="region">{`${address.localizedProvince}, `}</span>
                    )}
                    {address.postalCode && (
                        <span className="postal-code">{`${address.postalCode} / `}</span>
                    )}
                    {address.localizedCountry && (
                        <span className="country-name">{`${address.localizedCountry} `}</span>
                    )}
                </p>
            </div>
        </div>
    );
};

export function mapToStaticAddressProps(
    context: CheckoutContextProps,
    { address, type }: StaticAddressProps,
): WithCheckoutStaticAddressProps | null {
    const {
        checkoutState: {
            data: { getBillingCountries, getBillingAddressFields, getShippingAddressFields },
        },
    } = context;

    return {
        countries: getBillingCountries(),
        fields:
            type === AddressType.Billing
                ? getBillingAddressFields(address.countryCode)
                : type === AddressType.Shipping
                ? getShippingAddressFields(address.countryCode)
                : undefined,
    };
}

export default withCheckout(mapToStaticAddressProps)(memo(StaticAddress));
