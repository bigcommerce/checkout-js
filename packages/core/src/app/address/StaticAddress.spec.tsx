import {
    CheckoutSelectors,
    CheckoutService,
    createCheckoutService,
} from '@bigcommerce/checkout-sdk';
import { mount, render } from 'enzyme';
import React, { FunctionComponent } from 'react';

import { CheckoutProvider } from '../checkout';
import { getCountries } from '../geography/countries.mock';

import { getAddress } from './address.mock';
import AddressType from './AddressType';
import { getAddressFormFields } from './formField.mock';
import StaticAddress, { StaticAddressProps } from './StaticAddress';

describe('StaticAddress Component', () => {
    let StaticAddressTest: FunctionComponent<StaticAddressProps>;
    let checkoutService: CheckoutService;
    let checkoutState: CheckoutSelectors;
    let defaultProps: StaticAddressProps;

    beforeEach(() => {
        checkoutService = createCheckoutService();
        checkoutState = checkoutService.getState();

        defaultProps = {
            address: getAddress(),
        };

        jest.spyOn(checkoutState.data, 'getBillingCountries').mockReturnValue(getCountries());

        jest.spyOn(checkoutState.data, 'getBillingAddressFields').mockReturnValue(
            getAddressFormFields(),
        );

        jest.spyOn(checkoutState.data, 'getShippingAddressFields').mockReturnValue(
            getAddressFormFields(),
        );

        StaticAddressTest = (props) => (
            <CheckoutProvider checkoutService={checkoutService}>
                <StaticAddress {...props} />
            </CheckoutProvider>
        );
    });

    it('renders component with supplied props', () => {
        const tree = render(<StaticAddressTest {...defaultProps} />);

        expect(tree).toMatchSnapshot();
    });

    it('renders component when props are missing', () => {
        const tree = render(
            <StaticAddressTest
                {...defaultProps}
                address={{
                    ...defaultProps.address,
                    phone: '',
                    address2: '',
                }}
            />,
        );

        expect(tree).toMatchSnapshot();
    });

    it('renders component if required fields for billing address are not missing', () => {
        const container = mount(<StaticAddressTest {...defaultProps} type={AddressType.Billing} />);

        expect(checkoutState.data.getBillingAddressFields).toHaveBeenCalled();

        expect(container.find('.address-line-1').text()).toContain(defaultProps.address.address1);
    });

    it('does not render component if required fields for billing address are missing', () => {
        const container = mount(
            <StaticAddressTest
                {...defaultProps}
                address={{
                    ...defaultProps.address,
                    address1: '',
                }}
                type={AddressType.Billing}
            />,
        );

        expect(checkoutState.data.getBillingAddressFields).toHaveBeenCalled();

        expect(container.html()).toBe('');
    });

    it('renders component if required fields for shipping address are not missing', () => {
        const container = mount(
            <StaticAddressTest {...defaultProps} type={AddressType.Shipping} />,
        );

        expect(checkoutState.data.getShippingAddressFields).toHaveBeenCalled();

        expect(container.find('.address-line-1').text()).toContain(defaultProps.address.address1);
    });

    it('does not render component if required fields for shipping address are missing', () => {
        const container = mount(
            <StaticAddressTest
                {...defaultProps}
                address={{
                    ...defaultProps.address,
                    address1: '',
                }}
                type={AddressType.Shipping}
            />,
        );

        expect(checkoutState.data.getShippingAddressFields).toHaveBeenCalled();

        expect(container.html()).toBe('');
    });

    it('renders component if only custom fields are missing', () => {
        jest.spyOn(checkoutState.data, 'getBillingAddressFields').mockReturnValue([
            ...getAddressFormFields(),
            {
                custom: true,
                default: undefined,
                fieldType: 'text',
                id: 'foobar',
                label: 'Custom number',
                name: 'foobar',
                required: true,
                type: 'integer',
            },
        ]);

        const container = mount(
            <StaticAddressTest
                {...defaultProps}
                address={{
                    ...defaultProps.address,
                    customFields: [{ fieldId: 'foobar', fieldValue: '' }],
                }}
                type={AddressType.Billing}
            />,
        );

        expect(container.html()).not.toBe('');
    });
});
