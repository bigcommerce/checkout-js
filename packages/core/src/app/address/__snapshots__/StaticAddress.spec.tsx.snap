// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticAddress Component renders component when props are missing 1`] = `
<div
  class="vcard checkout-address--static"
>
  <p
    class="fn address-entry"
  >
    <span
      class="first-name"
    >
      Test 
    </span>
    <span
      class="family-name"
    >
      Tester
    </span>
  </p>
  <p
    class="address-entry"
  >
    <span
      class="company-name"
    >
      Bigcommerce 
    </span>
    <span
      class="tel"
    />
  </p>
  <div
    class="adr"
  >
    <p
      class="street-address address-entry"
    >
      <span
        class="address-line-1"
      >
        12345 Testing Way 
      </span>
    </p>
    <p
      class="address-entry"
    >
      <span
        class="locality"
      >
        Some City, 
      </span>
      <span
        class="region"
      >
        California, 
      </span>
      <span
        class="postal-code"
      >
        95555 / 
      </span>
      <span
        class="country-name"
      >
        United States 
      </span>
    </p>
  </div>
</div>
`;

exports[`StaticAddress Component renders component with supplied props 1`] = `
<div
  class="vcard checkout-address--static"
>
  <p
    class="fn address-entry"
  >
    <span
      class="first-name"
    >
      Test 
    </span>
    <span
      class="family-name"
    >
      Tester
    </span>
  </p>
  <p
    class="address-entry"
  >
    <span
      class="company-name"
    >
      Bigcommerce 
    </span>
    <span
      class="tel"
    >
      555-555-5555
    </span>
  </p>
  <div
    class="adr"
  >
    <p
      class="street-address address-entry"
    >
      <span
        class="address-line-1"
      >
        12345 Testing Way 
      </span>
    </p>
    <p
      class="address-entry"
    >
      <span
        class="locality"
      >
        Some City, 
      </span>
      <span
        class="region"
      >
        California, 
      </span>
      <span
        class="postal-code"
      >
        95555 / 
      </span>
      <span
        class="country-name"
      >
        United States 
      </span>
    </p>
  </div>
</div>
`;
